### PR TITLE
Add output detach for Canberra metric

### PR DESCRIPTION
**Pull request by KickItLikeShika**
Add `output.detach()` in `CanberraMetric`

Check list:

- [ ] New tests are added (if a new feature is added)
- [ ] New doc strings: description and/or example code are in RST format
- [ ] Documentation is updated (if required)

### CODE REVIEW

**Review by vfdev-5 — APPROVED** [review]
Thanks @KickItLikeShika !
